src/storage/store/engine.rs

```diff
@@ -148,6 +148,7 @@ struct TransactionCounts {
     system_messages: u64,
 }
 
+#[derive(Clone)]
 pub struct ShardEngine {
```

Comment:
Hmm, I'm not entirely sure if this is safe to clone, there might be some weirdness with the trie in-memory state. I think it's better to construct a new instance. This is what we're doing with the SnapchainNode.

Reply:
Done, this was only used in tests but I worked around it. Just want to pass the same db instance around.